integration/connector_test.go

```diff
@@ -0,0 +1,430 @@
+	if err == nil {
+		err = dittoClient.Connect()
+	}
+
+	if err != nil {
+		mqttClient.Disconnect(uint(cfg.MqttQuiesceMs))
+		require.NoError(suite.T(), err, "initialize ditto client")
+	}
+
+	feature := &model.Feature{}
+	feature.WithProperty(propertyName, "testValue")
+
+	cmd := things.NewCommand(model.NewNamespacedIDFrom(thingCfg.DeviceID)).Twin().Feature(featureName).
+		Modify(feature)
+	msg := cmd.Envelope(protocol.WithResponseRequired(false))
+
+	err = dittoClient.Send(msg)
+	require.NoError(suite.T(), err, "create test feature")
+
+	dittoClient.Subscribe(func(requestID string, msg *protocol.Envelope) {
+		if msg.Path != fmt.Sprintf("/features/%s/inbox/messages/%s", featureName, commandName) {
+			suite.T().Logf("unexpected command: %s", msg.Path)
+			return
+		}
+
+		headers := protocol.NewHeaders(protocol.WithCorrelationID(msg.Headers.CorrelationID()),
+			protocol.WithContentType("application/json"))
+
+		value, ok := msg.Value.(string)
+		if !ok {
+			suite.T().Errorf("unexpected message payload: %v, %T", msg.Value, msg.Value)
+			return
+		}
+
+		response := fmt.Sprintf(commandResponseFormat, value)
+
+		reply := &protocol.Envelope{
+			Topic:   msg.Topic,
+			Headers: headers,
+			Path:    strings.Replace(msg.Path, "/inbox/", "/outbox/", 1),
+			Value:   response,
+			Status:  http.StatusNoContent,
+		}
+
+		if err := dittoClient.Reply(requestID, reply); err != nil {
+			suite.T().Error("failed to send response:", err)
+		}
+	})
+
+	suite.dittoClient = dittoClient
+	suite.mqttClient = mqttClient
+	suite.cfg = cfg
+	suite.thingCfg = thingCfg
+
+	suite.thingURL = fmt.Sprintf("http://%s:%d/api/2/things/%s", cfg.DittoHost, cfg.DittoPort, thingCfg.DeviceID)
+	suite.featureURL = fmt.Sprintf("%s/features/%s", suite.thingURL, featureName)
+}
+
+func (suite *ConnectorSuite) TearDownSuite() {
+	if _, err := suite.doRequest(http.MethodDelete, suite.featureURL); err != nil {
+		suite.T().Logf("error while deleting test feature: %v", err)
+	}
+
+	suite.dittoClient.Disconnect()
+	suite.mqttClient.Disconnect(uint(suite.cfg.MqttQuiesceMs))
+}
+
+func TestConnectorSuite(t *testing.T) {
+	suite.Run(t, new(ConnectorSuite))
+}
+
+func (suite *ConnectorSuite) TestConnectionStatus() {
+	statusURL := fmt.Sprintf("%s/features/ConnectionStatus/properties/status", suite.thingURL)
+	body, err := suite.doRequest("GET", statusURL)
+	require.NoError(suite.T(), err, "connection status property should be available")
+
+	type connectionStatus struct {
+		ReadySince time.Time `json:"readySince"`
+		ReadyUntil time.Time `json:"readyUntil"`
+	}
+
+	status := &connectionStatus{}
+	err = json.Unmarshal(body, status)
+	require.NoError(suite.T(), err, "connection status should be parsed")
+
+	suite.T().Logf("%+v", status)
+	suite.T().Logf("current time: %v", time.Now())
+
+	delta := int64(suite.cfg.TimeDeltaMs)
+	assert.Less(suite.T(), status.ReadySince.UnixMilli(), time.Now().UnixMilli()+delta, "readySince should be BEFORE current time")
```

Comment:
In some cases when the suite connector is restarted - e.g. to reload its configuration - and the test is started right away, it is possible for readySince to be reported as few milliseconds after time.Now(). The delta prevents the test from failing in such occurrences.

Reply:
I think we should keep this code because the connection status only contains "ready since" and "ready until". This algorithm is a good sanity check for "ready since".

Reply:
@dimiter-todorov-georgiev - initially, there was no loop to check that the device is online and so called `sanity check` was an issue. The reason was very simple, it's possible to have a running TC but offline device i.e. most probably all assertions will fail. This was fixed with the outer loop and I guess that the delta is not needed any more or I am missing something?

Reply:
I think it's more of a way to have some wiggle room for the client and server having slightly different clocks. In my mind, this is like comparing floats, you always want to have some epsilon instead of comparing directly, due to inaccuracies.

Reply:
If that the case, I could agree but some details have to be clarified. Why are 5 sec used as a default value? I guess that by default the delta should be 0 sec and if there is clock diff, it can be specified. Does it make sense?

Reply:
I think I understand. We can make this value configurable e.g. through an environment variable.

Reply:
What's the status here? Now, there is an environment variable but the default value is still 5 sec. Are we going to set it to 0 sec?